packages/xo-server/src/xo-mixins/remotes.js

```diff
@@ -83,7 +83,15 @@ export default class {
 
   async testRemote(remote) {
     const handler = await this.getRemoteHandler(remote)
-    return handler.test()
+    const answer = await handler.test()
+
+    if (answer.success === true) {
+      const benchmark = { ...answer }
+      delete benchmark.success
+      await this.updateRemote(remote, { benchmark })
```

Comment:
I would say the `testRemote` add the `benchmark` to the `remote` benchmarks and update the `remote`.